src/libkernel/sscp/ptx/integer.cpp

```diff
@@ -38,3 +38,19 @@ HIPSYCL_SSCP_BUILTIN __hipsycl_int32 __hipsycl_sscp_mul24_s32(__hipsycl_int32 a,
 HIPSYCL_SSCP_BUILTIN __hipsycl_uint32 __hipsycl_sscp_mul24_u32(__hipsycl_uint32 a, __hipsycl_uint32 b) {
   return __nv_umul24(a, b);
 }
+
+extern "C" __hipsycl_int32 __nv_clz(__hipsycl_int32);
+extern "C" __hipsycl_int64 __nv_clzll(__hipsycl_int64);
```

Comment:
Do we actually need that declaration here? I think you could just include https://github.com/OpenSYCL/OpenSYCL/blob/develop/include/hipSYCL/sycl/libkernel/sscp/builtins/ptx/libdevice.hpp
which contains declarations for the full functionality of NVIDIA libdevice, including clz.